src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -12603,6 +12609,23 @@ void WaterThermalTankData::oneTimeInit(EnergyPlusData &state)
     }
 }
 
+void WaterThermalTankData::setBackupElementCapacity(EnergyPlusData &state)
+{
+    // Fix for #9001: The BackupElementCapacity was not being reset from the autosize value (-99999) which resulted in
+    // negative electric consumption.  Using a test for any negative numbers here instead of just -99999 for safety.
+    // Only reset the backup element capacity if a problem has been occured.
+    if (this->HeatPumpNum > 0) {
+        if (state.dataWaterThermalTanks->HPWaterHeater(this->HeatPumpNum).TypeNum == DataPlant::TypeOf_HeatPumpWtrHeaterWrapped) return;
+        if (state.dataWaterThermalTanks->HPWaterHeater(this->HeatPumpNum).BackupElementCapacity < 0.0) {
+            state.dataWaterThermalTanks->HPWaterHeater(this->HeatPumpNum).BackupElementCapacity = this->MaxCapacity;
+        }
+    } else if (this->DesuperheaterNum > 0) {
+        if (state.dataWaterThermalTanks->WaterHeaterDesuperheater(this->DesuperheaterNum).BackupElementCapacity < 0.0) {
+            state.dataWaterThermalTanks->WaterHeaterDesuperheater(this->DesuperheaterNum).BackupElementCapacity = this->MaxCapacity;
+        }
+    }
+}
+
```

Comment:
@RKStrand I posted this comment here since I thought updating the HP and desuperheater backup element capacity may not be quite correct since the backup resistive element is in the tank. Writing out this comment and searching through code makes me think this is the correct change.

I pulled this branch to review. I was looking for a backup element and that input does not exist. It seems it is an inferred parameter for the models that interact with tanks. But it is not even used in the tank model (or maybe it does and I can't see it on cursory review). It is used in the models that interact with the tank. The HP and desuperheater pull this back-door variable from the tanks. This review shows:

1) there are no diffs now.
2) When these other models pull the tank max capacity into a backup element variable, and that max capacity was autosized, this change now updates those values correctly (after the tank capacity is autosized) from -9999 to whatever the tank capacity sizes to. I agree with what you did here.
3) If this backup element is used for something within a model, the corner case of its use is not exercised in an example file, otherwise there would be eio diffs (hard-sized coil capacities do not report to the eio, and changes in tank temp would result in plant diffs).

![image](https://user-images.githubusercontent.com/5778156/133710383-c7012f11-398f-4589-b061-d01caed8ee5a.png)

The mixed tank has a single max capacity. This model, I would think, would have a backup element of the same capacity. The stratified tank has 2 capacity inputs (resistive elements). That model should NOT have a backup element (backup capacity = 0 since there are already 2 elements). So without rethinking the model intent I have to agree with what you have done for this specific issue.

    WaterHeater:Mixed,
      N4 , \field Heater Maximum Capacity

    WaterHeater:Stratified,
      N6 , \field Heater 1 Capacity
      N9 , \field Heater 2 Capacity